src/main/java/org/testng/internal/Systematiser.java

```diff
@@ -0,0 +1,96 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+
+import java.util.Comparator;
+
+public final class Systematiser {
```

Comment:
@cbeust - I have added the required javadocs as well. Can you please see if thats fine ?